src/libraries/System.Text.RegularExpressions/gen/UpgradeToRegexGeneratorAnalyzer.cs

```diff
@@ -198,7 +198,12 @@ private static bool TryValidateParametersAndExtractArgumentIndices(ImmutableArra
             for (int i = 0; i < arguments.Length; i++)
             {
                 IArgumentOperation argument = arguments[i];
-                string argumentName = argument.Parameter.Name;
+                // argument.Parameter can be null for __arglist
```

Comment:
I removed the check. But I'm feeling this is still reachable. There is nothing that guarantees the method we're looking for has the signature we're expecting. One can create his own `System.Text.RegularExpressions.Regex` type and define methods with the same name, but with `__arglist` parameter. This is very uncommon though.

Reply:
I believe that we wouldn't hit this line even if someone defined their own Regex type, since we do a symbol equality comparison to make sure the target type is the framework's Regex type.

Reply:
@joperezr Nothing seems to ensure this type is coming from CorLib. We first called `GetMetadataTypeByName("System.Text.RegularExpressions.Regex")` to get regex type (which can return user-defined type) then we searched for methods with some given names. Then we're iterating over arguments of invocations of these methods.

Anything I'm missing?

Reply:
You are right. I'd still think that will be rare enough that is probably not something we want to do, especially since it adds an if-check for every method we analyze, so I'm more leaning towards keeping this as it was.